src/api_server/src/session.rs

```diff
@@ -104,13 +113,63 @@ impl<'a> Session<'a> {
         self.is_authenticated() && !self.is_expired()
     }
 
+    pub fn attestation_results(&self) -> Option<AttestationResults> {
+        self.attestation_results.clone()
+    }
+
     pub fn set_attestation_results(&mut self, attestation_results: AttestationResults) {
         self.attestation_results = Some(attestation_results)
     }
 
     pub fn set_tee_public_key(&mut self, key: TeePubKey) {
         self.tee_pub_key = Some(key)
     }
+
+    pub fn to_jwe(&self, payload_data: Vec<u8>) -> Result<Response> {
+        let tee_pub_key = self
+            .tee_public_key()
+            .ok_or_else(|| anyhow!("No TEE public Key"))?;
+        if tee_pub_key.alg != *RSA_ALGORITHM {
+            return Err(anyhow!("Unsupported TEE Pub Key type or algorithm"));
```

Comment:
Please change this into `bail!()` macro to make it dry
```suggestion
            bail!("Unsupported TEE Pub Key type or algorithm");
```